### PR TITLE
Bootstrap

### PR DESCRIPTION
**Pull request by hannesvdvreken**
### Fixed
- Requiring autoloader from index script in public folder
### Changed
- No more `$app` in global namespace

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -55,7 +55,7 @@ $app = (new Glue())
         SomeProvider::class,
         AnotherProvider::class,
         LeagueRouteServiceProvider::class
-    ]);
```
Comment (Anahkiasen) [inline README.md:58]: ouch

File: src/Console/Commands/BootstrapCommand.php [hunk 2]
```diff
@@ -74,16 +74,20 @@ protected function execute(InputInterface $input, OutputInterface $output)
...
 <?php
-require 'vendor/autoload.php';
+require '../vendor/autoload.php';
 
-$app = new Madewithlove\Glue\Glue();
-$app->run();
+call_user_func(function () {
+    $app = new Madewithlove\Glue\Glue();
+    $app->run();
+});
```
Comment (Anahkiasen) [inline src/Console/Commands/BootstrapCommand.php:90]: Do JS style self executing closures work in PHP? Like `function (){}()` ?

Comment (hannesvdvreken) [inline src/Console/Commands/BootstrapCommand.php:90]: YES 🙌

in PHP 7 😔

Comment (hannesvdvreken) [inline src/Console/Commands/BootstrapCommand.php:90]: it's not documented but it is seen in action here: http://php.net/manual/en/migration70.new-features.php#migration70.new-features.generator-return-expressions

**Comment by Anahkiasen** [comment]
In this case the index.php should literally be the whole app so it shouldn't matter if it's there no?

**Comment by hannesvdvreken** [comment]
but this takes away the possibility to do

```
global $app;

$app->...
```

in any class/function anywhere

**Comment by Anahkiasen** [comment]
No yeah I know, my reasoning is more that this is not the kind of package you can use without knowing a minimum your shit, so what are the odds that someone using Glue, PSR11 and PSR7 and all, would also be someone sprinkling `global`s everywhere :D

**Comment by hannesvdvreken** [comment]
Your call :smile:

**Comment by Anahkiasen** [comment]
Aduno, on the fence. One the one hand it's better, on the other hand I don't want to make the initial bootstrap more complicated, like, make people think this is _required_. Will think about this a bit, can you rebase the PR to latest master?

**Comment by hannesvdvreken** [comment]
Rebased

**Comment by hannesvdvreken** [comment]
I removed my clone with these commits in the meantime so this PR became useless

**Pull request closed without merging**